src/components/GameOfLife/gameOfLife.test.js

```diff
@@ -0,0 +1,365 @@
+import React from 'react';
+import { render, fireEvent } from '@testing-library/react'
+import '@testing-library/jest-dom/extend-expect'
+import GameOfLife, {next, countLiveNeighbours, generateRandomBoard, areIterationsIdentical } from './gameOfLife';
+
+describe('Test countLiveNeighbours', () => {
+  test('return 8 when all cells are live', () => {
+    const input = [[1, 1, 1], [1, 1, 1], [1, 1, 1]];
+    const expectedOutput = 8;
+    expect(countLiveNeighbours(input, 1, 1)).toBe(expectedOutput);
+  });
+  test('return 3 for a corner cell', () => {
+    const input = [
+      [0, 1, 1],
+      [1, 1, 1],
+      [1, 1, 1],
+    ];
+    const expectedOutput = 3;
+    expect(countLiveNeighbours(input, 0, 0)).toBe(expectedOutput);
+  });
+  test('return 3 for a corner cell with non-square grid', () => {
+    const input = [
+      [0, 1, 1],
+      [1, 1, 1],
+      [1, 1, 1],
+      [1, 1, 1],
+    ];
+    const expectedOutput = 3;
+    expect(countLiveNeighbours(input, 3, 2)).toBe(expectedOutput);
+  });
+});
+
+const getNeighbours = (grid, i, j) => {
+  const neighbours = [];
+
+  for (let x = -1; x < 2; x += 1) {
+    for (let y = -1; y < 2; y += 1) {
+      if (grid[i + x] !== undefined && grid[i + x][j + y] !== undefined && (x !== 0 || y !== 0)) {
+        neighbours.push([[i + x], [j + y]]);
+      }
+    }
+  }
+
+  return neighbours;
+};
+
+const generateLiveNeighbours = (grid, i, j, liveNeighbours) => {
+  const newGrid = grid;
+  const neighbours = getNeighbours(grid, i, j);
+  if (liveNeighbours > neighbours.length) {
+    return null;
+  }
+  for (let x = 0; x < liveNeighbours; x += 1) {
+    newGrid[neighbours[x][0]][neighbours[x][1]] = 1;
+  }
+
+  return newGrid;
+};
+
+const generateTestData = (size, i, j, liveNeighbours, status) => {
+  const grid = Array(size).fill(0).map(() => Array(size).fill(0));
+
+  grid[i][j] = status;
+
+  return generateLiveNeighbours(grid, i, j, liveNeighbours);
+};
```

Comment:
DRY is cool but not for tests IMO.

1. nobody knows from the test itself, what is the input
2. are you sure this function is implemented correctly?

I think copy&paste of the data for each test scenario (even when duplicated) would be a better approach.